File: LeetCode/0242. Valid Anagram/Solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        if len(s) != len(t):
            return False

        letters_counter = {}

        for s_char, t_char in zip(s, t):
            if s_char in letters_counter:
                letters_counter[s_char] += 1
            else:
                letters_counter[s_char] = 1

            if t_char in letters_counter:
                letters_counter[t_char] -= 1
            else:
                letters_counter[t_char] = -1

        for key in letters_counter:
            if letters_counter[key] != 0:
                return False

        return True

File: LeetCode/0242. Valid Anagram/Solution.java
class Solution {
    public boolean isAnagram(String s, String t) {
        
        if (s.length() != t.length()) {
            return false;
        }

        HashMap<Character, Integer> lc =  new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            Character s_char = s.charAt(i);
            Character t_char = t.charAt(i);

            lc.putIfAbsent(s_char, 0);
            lc.putIfAbsent(t_char, 0);

            lc.put(s_char, lc.get(s_char)+1);
            lc.put(t_char, lc.get(t_char)-1);
        }
        
        for (Integer v: lc.values()) {
            if (v != 0) {
                return false;
            }
        }

        return true;
    }
}
